orbit/pkg/table/filevault_prk/filevault_prk_darwin.go

```diff
@@ -0,0 +1,36 @@
+//go:build darwin
+// +build darwin
+
+package filevault_prk
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("base64_encrypted"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+//
+// Constraints for generating can be retrieved from the queryContext.
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	encryptedKey, err := os.ReadFile("/var/db/FileVaultPRK.dat")
+	if err != nil {
+		// TODO: What if the file does not exist? Should we check for "open /var/db/FileVaultPRK.dat:
+		// no such file or directory" and return an empty result instead?
```

Comment:
I think so! this also addresses your other TODO in `server/service/osquery_utils/queries.go`

Reply:
Would deleting the prior key from the datastore render the disk unrecoverable in certain circumstances?

Reply:
> Would deleting the prior key from the datastore render the disk unrecoverable in certain circumstances?

I'm guessing you're thinking of an edge case? something like: we fetched the key in the past, but the user removes the file? could you give us more details?

More generally, `/var/db/FileVaultPRK.dat` is the encrypted value of the recovery key, but nothing else. It will affect Fleet's ability to escrow the recovery key.

Reply:
> I'm guessing you're thinking of an edge case? something like: we fetched the key in the past, but the user removes the file? could you give us more details?

Yes, that's my concern. To rephrase it, under what circumstances should we be deleting or destructively updating `host_disk_encryption_keys` table when there is a prior value stored?

Reply:
I think it's okay for now to overwrite the current value with whatever we get. If I'm not mistaken, this is the way it currently works.

The file is automatically removed when disk encryption is turned off, or you rotate the key without escrow, etc.

In the future, we could try to be smart and detect if the user manually removed the key but it's going to be tricky.

To add more color, the `SecurityInfo` MDM command also returns the contents of the file, but only if the file exists, if you go and remove the file manually, the value comes back empty (I have tried). My guess is that the mdm client reads this file too.